modules/auxiliary/scanner/smb/smb_enumshares.rb

```diff
@@ -310,87 +282,61 @@ def get_files_info(ip, _rport, shares, info)
+          connect(versions: [1])
+        else
+          connect(versions: [1, 2, 3])
+        end
         smb_login
-        shares = smb_netshareenumall
-
-        os_info = get_os_info(ip, rport)
+        shares = simple.client.net_share_enum_all(ip)
+        os_info = get_os_info(ip)
         print_status(os_info) if os_info
 
         if shares.empty?
-          print_status('No shares collected')
+          print_status('No shares available')
         else
-          shares_info = shares.map { |x| "#{x[0]} - (#{x[1]}) #{x[2]}" }.join(', ')
-          shares_info.split(', ').each do |share|
-            print_good share
+          shares.each do |share|
+            print_good("#{share[:name]} - (#{share[:type]}) #{share[:comment]}")
           end
+
+          # Map RubySMB shares to the same data format as it was with Rex SMB
+          report_shares = shares.map { |share| [share[:name], share[:type], share[:comment]] }
           report_note(
             host: ip,
             proto: 'tcp',
             port: rport,
             type: 'smb.shares',
-            data: { shares: shares },
+            data: { shares: report_shares },
             update: :unique_data
           )
 
           if datastore['SpiderShares']
-            begin
-              connect(versions: [1])
-              smb_login
-              get_files_info(ip, rport, shares, info)
-            rescue ::Rex::Proto::SMB::Exceptions::Error, Errno::ECONNRESET => e
-              print_error(
-                "Error when Spidering shares recursively (#{e}). This feature "\
-                'is only available with Rex client (SMB1 only) and the host '\
-                "probably doesn't support SMB1."
-              )
-            end
+            get_files_info(ip, shares)
           end
-
-          break if rport == 139
         end
       rescue ::Interrupt
         raise $ERROR_INFO
-      rescue ::Rex::Proto::SMB::Exceptions::LoginError,
-             ::Rex::Proto::SMB::Exceptions::ErrorCode => e
-        print_error(e.message)
-        return if e.message =~ /STATUS_ACCESS_DENIED/
-      rescue Errno::ECONNRESET,
-             ::Rex::Proto::SMB::Exceptions::InvalidType,
-             ::Rex::Proto::SMB::Exceptions::ReadPacket,
-             ::Rex::Proto::SMB::Exceptions::InvalidCommand,
-             ::Rex::Proto::SMB::Exceptions::InvalidWordCount,
-             ::Rex::Proto::SMB::Exceptions::NoReply => e
+      rescue Errno::ECONNRESET => e
         vprint_error(e.message)
-        next if !shares.empty? && (rport == 139) # no results, try again
       rescue Errno::ENOPROTOOPT
         print_status('Wait 5 seconds before retrying...')
         select(nil, nil, nil, 5)
         retry
-      rescue ::Exception => e
-        next if e.to_s =~ /execution expired/
-        next if !shares.empty? && (rport == 139)
-
+      rescue Rex::ConnectionTimeout => e
+        print_error e.to_s
+        return
+      rescue Rex::Proto::SMB::Exceptions::LoginError => e
```

Comment:
Consider adding a `rescue` for any possible RubySMB exceptions that might not have been caught above.